### PR TITLE
Test aspects

### PR DESCRIPTION
**Pull request by mmozerwayfair**
Adding test for _aspects.py

### CODE REVIEW

**Comment by mmozerwayfair** [comment]
The test will only work if spacys language pack is actually installed and this is not something that can be trivially worked around.

In the in-house package, the language pack is installed during the build of the docker image.
Not exactly sure where to do that here.
.github/actions/install-dependencies/action.sh
Will probably work, but wanted to check before messing with that file

File: .github/actions/install-dependencies/action.sh [hunk 1]
```diff
@@ -6,6 +6,7 @@ python -m pip install --upgrade pip
 if [[ "${INSTALL_REQUIREMENTS}" == "true"  ]]; then
   echo "Installing code requirements"
   pip install -r requirements.txt
+  python -m spacy download en_core_web_sm #download spacy language pack
```
Comment (chrisantonellis) [inline .github/actions/install-dependencies/action.sh:9]: lets do this in the `docker/Dockerfile` instead as the resources are needed beyond just the test / lint suite 👍

File: tests/test_aspects.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import pandas as pd
```
Comment (chrisantonellis) [inline tests/test_aspects.py:1]: nice tests!

Comment (chrisantonellis) [reply tests/test_aspects.py:1]: very clean!

**Comment by chrisantonellis** [comment]
Regarding installing the `spaCy` resources on CI, I would add a stage to the `main` workflow here: https://github.com/wayfair-incubator/extra-model/blob/test_aspects/.github/workflows/main.yml#L109

let me know if that makes sense! thanks!

**Comment by mmozerwayfair** [comment]
Had to add spacy and one of its dependencies to the requirements file, but now the tests pass.

File: extra_model/_aspects.py [hunk 3]
```diff
@@ -16,8 +16,10 @@
 def compound_noun_list(token):
     """
     Find compound nouns
-    :param token: (spacy.token): token for which to generate potential compound nouns
-    :return nouns ([string]): list of potential compounds
+    :param token: token for which to generate potential compound nouns
+    :type token: :class:`spacy.token`
+    :return: list of potential compounds
+    :rtyoe: [string]
```
Comment (romatik) [inline extra_model/_aspects.py:22]: typo

**Review by romatik — APPROVED** [review]

**Pull request merged**